go/libraries/doltcore/sqle/binlogreplication/binlog_position_store.go

```diff
@@ -133,3 +127,18 @@ func (store *binlogPositionStore) Delete(ctx *sql.Context) error {
 
 	return filesys.Delete(filepath.Join(binlogPositionDirectory, binlogPositionFilename), false)
 }
+
+// createDoltCfgDir creates the .doltcfg directory if it doesn't already exist.
+func createDoltCfgDir(filesys filesys.Filesys) error {
```

Comment:
Seems like this might be worth moving to a common init service? We kind of always want this to exist for a running server.

Reply:
Started digging into this and realized that the binlog code isn't using the configured `.doltcfg` directory. I'm going to fix that and add the create dir init service in a new PR.